skore-ui/src/components/DraggableList.vue

```diff
@@ -0,0 +1,317 @@
+
+function topDropIndicatorStyles() {
+  if (dropIndicatorPosition.value === -1) {
+    return {
+      marginTop: `${movingItemHeight.value}px`,
+      marginBottom: "var(--spacing-gap-normal)",
+    };
+  }
+  return {};
+}
+
+function dropIndicatorStyles(index: number) {
+  const y = dropIndicatorPosition.value === index ? movingItemHeight.value : 0;
+  if (direction === "up") {
+    return {
+      marginTop: `calc(var(--spacing-gap-normal) + ${y}px)`,
+    };
+  } else if (direction === "down") {
+    return {
+      marginTop: `var(--spacing-gap-normal)`,
+      marginBottom: `${y}px`,
+    };
+  }
+}
+
+function capturedStyles() {
+  const a = -4 * (Math.PI / 180);
+  const containerBounds = container
+    .value!.querySelector(".content-wrapper")!
+    .getBoundingClientRect();
+  const width = containerBounds.width;
+  const rotatedWidth = width * Math.cos(a) + movingItemHeight.value * Math.sin(a);
+  const ratio = rotatedWidth / width;
+
+  return {
+    transform: `translateY(${movingItemY.value}px) rotate(${a}rad) scale(${ratio})`,
+  };
+}
+
+function makeModifier() {
+  const containerBounds = container.value!.getBoundingClientRect();
+
+  return interact.modifiers.restrict({
+    restriction: (x, y, { element }) => {
+      const content = element?.parentElement?.parentElement?.querySelector(".content");
+      const { height } = content?.getBoundingClientRect() ?? { height: 0 };
+      return {
+        top: containerBounds?.top - height,
+        right: containerBounds?.right,
+        bottom: containerBounds?.bottom + height,
+        left: containerBounds?.left,
+      };
+    },
+  });
+}
+
+onMounted(() => {
+  if (props.autoScrollContainerSelector !== undefined) {
+    const element = document.querySelector(props.autoScrollContainerSelector);
+    if (element) {
+      const isSimplebar = element.hasAttribute("data-simplebar");
+      if (isSimplebar) {
+        const sb = new Simplebar(element as HTMLElement);
+        const scrollElement = sb.getScrollElement();
+        sb.unMount();
+        if (scrollElement) {
+          autoScrollContainer = scrollElement;
+        }
+      } else {
+        autoScrollContainer = element as HTMLElement;
+      }
+    }
+  } else {
+    autoScrollContainer = container.value!.parentElement ?? document.body;
+  }
+
+  interactable = interact(".handle").draggable({
+    autoScroll: {
+      enabled: true,
+      container: autoScrollContainer,
+    },
+    startAxis: "y",
+    lockAxis: "y",
+    modifiers: [makeModifier()],
+    listeners: {
+      async start(event) {
+        // make a rasterized copy of the moving element
+        const content = event.target.parentElement.querySelector(".content");
+        if (content && container.value) {
+          const parentBounds = container.value.getBoundingClientRect();
+          const bounds = content.getBoundingClientRect();
+
+          movingItemAsPngData.value = await toPng(content);
```

Comment:
Sorry, I surely miss something here. Why is it needed to raster the component manually? Isn't it the purpose of the specialized `CacheableComponent`?

Reply:
This one is the one that follows the cursor.